CMakeLists.txt

```diff
@@ -0,0 +1,260 @@
+    pure-data/src/m_glob.c
+    pure-data/src/m_imp.h
+    pure-data/src/m_memory.c
+    pure-data/src/m_obj.c
+    pure-data/src/m_pd.c
+    pure-data/src/m_pd.h
+    pure-data/src/m_sched.c
+    pure-data/src/s_audio.c
+    pure-data/src/s_audio_dummy.c
+    pure-data/src/s_inter.c
+    pure-data/src/s_loader.c
+    pure-data/src/s_main.c
+    pure-data/src/s_path.c
+    pure-data/src/s_print.c
+    pure-data/src/s_stuff.h
+    pure-data/src/s_utf8.c
+    pure-data/src/s_utf8.h
+    pure-data/src/x_acoustics.c
+    pure-data/src/x_arithmetic.c
+    pure-data/src/x_array.c
+    pure-data/src/x_connective.c
+    pure-data/src/x_gui.c
+    pure-data/src/x_interface.c
+    pure-data/src/x_list.c
+    pure-data/src/x_midi.c
+    pure-data/src/x_misc.c
+    pure-data/src/x_net.c
+    pure-data/src/x_scalar.c
+    pure-data/src/x_text.c
+    pure-data/src/x_time.c
+    pure-data/src/x_vexp.c
+    pure-data/src/x_vexp.h
+    pure-data/src/x_vexp_fun.c
+    pure-data/src/x_vexp_if.c
+)
+
+set(PD_EXTRA_SOURCES
+    pure-data/extra/bob~/bob~.c
+    pure-data/extra/bonk~/bonk~.c
+    pure-data/extra/choice/choice.c
+    pure-data/extra/fiddle~/fiddle~.c
+    pure-data/extra/loop~/loop~.c
+    pure-data/extra/lrshift~/lrshift~.c
+    pure-data/extra/pique/pique.c
+    pure-data/extra/sigmund~/sigmund~.c
+    pure-data/extra/stdout/stdout.c
+)
+
+set(LIBPD_SOURCES
+    libpd_wrapper/s_libpdmidi.c
+    libpd_wrapper/x_libpdreceive.c
+    libpd_wrapper/x_libpdreceive.h
+    libpd_wrapper/z_hooks.c
+    libpd_wrapper/z_hooks.h
+    libpd_wrapper/z_libpd.c
+)
+set(LIBPD_UTILS_SOURCES
+    libpd_wrapper/util/ringbuffer.c
+    libpd_wrapper/util/ringbuffer.h
+    libpd_wrapper/util/z_print_util.c
+    libpd_wrapper/util/z_print_util.h
+    libpd_wrapper/util/z_queued.c
+    libpd_wrapper/util/z_queued.h
+)
+
+source_group(pd FILES ${PD_SOURCES})
+source_group(pdextra FILES ${PD_EXTRA_SOURCES})
+source_group(libpd FILES ${LIBPD_SOURCES})
+source_group(libpdutils FILES ${LIBPD_UTILS_SOURCES})
+
+include_directories(libpd_wrapper)
+include_directories(pure-data/src)
+
+find_package(Threads REQUIRED)
+
+# create final list of source files
+set(SOURCE_FILES ${PD_SOURCES} ${LIBPD_SOURCES})
+if(PD_UTILS)
+    list(APPEND SOURCE_FILES ${LIBPD_UTILS_SOURCES})
+endif()
+if(PD_EXTRA)
+    list(APPEND SOURCE_FILES ${PD_EXTRA_SOURCES})
+endif()
+
+add_library(libpd_static STATIC ${SOURCE_FILES})
+add_library(libpd        SHARED ${SOURCE_FILES})
+
+# set the output library name for libpd depending on the settings
+set(LIBPD_OUTPUT_NAME     pd)
+if (WIN32)
+    set(LIBPD_OUTPUT_NAME libpd)
+endif()
+if (PD_MULTI)
+    set(LIBPD_OUTPUT_NAME ${LIBPD_OUTPUT_NAME}-multi)
+endif()
+if (WIN32)
+    set_target_properties(libpd_static PROPERTIES OUTPUT_NAME ${LIBPD_OUTPUT_NAME}-static)
+else()
+    set_target_properties(libpd_static PROPERTIES OUTPUT_NAME ${LIBPD_OUTPUT_NAME})
+endif()
+set_target_properties(libpd        PROPERTIES OUTPUT_NAME ${LIBPD_OUTPUT_NAME})
```

Comment:
Should this line have such a large whitespace gap in the middle?

Reply:
Things are aligned with whitespace for readability. Actually, a few spaces are missing in this line.